src/components/modals/ManageAntModal/ManageAntModal.tsx

```diff
@@ -0,0 +1,264 @@
+import { useEffect, useRef, useState } from 'react';
+
+import { useIsMobile } from '../../../hooks';
+import { useGlobalState } from '../../../state/contexts/GlobalState';
+import { ArweaveTransactionID } from '../../../types';
+import { ANTContractState } from '../../../types';
+import { NotebookIcon, PencilIcon } from '../../icons';
+import CopyTextButton from '../../inputs/buttons/CopyTextButton/CopyTextButton';
+import TransactionStatus from '../../layout/TransactionStatus/TransactionStatus';
+import RowItem from '../../layout/tables/RowItem/RowItem';
+
+function ManageAntModal({
+  contractId,
+  setShowModal,
+  state,
+  confirmations,
+  targetId,
+}: {
+  contractId: ArweaveTransactionID;
+  setShowModal: (show: boolean) => void;
+  state: ANTContractState;
+  confirmations: number;
+  targetId?: string;
+}) {
+  const [{ arnsSourceContract }] = useGlobalState();
+  const modalRef = useRef(null);
+  const [antDetails, setAntDetails] = useState<any[]>();
+  const isMobile = useIsMobile();
+  // todo: manage asset modal writes asset modifications to contract. It will auto detect if the asset is an ANT, name, or undername.
+  // if the asset is a name, it will write modifications to the registry. If its an undername, it will write mods to the ant. If its an ant, it will write mods to the ant.
+
+  useEffect(() => {
+    setDetails();
+  }, [contractId, state, targetId, confirmations]);
+
+  function handleClickOutside(e: any) {
+    if (modalRef.current && modalRef.current === e.target) {
+      setShowModal(false);
+    }
+    return;
+  }
+
+  function getAssociatedNames(txId: ArweaveTransactionID) {
+    return Object.entries(arnsSourceContract.records)
+      .map(([name, id]) => {
+        if (id === txId.toString()) return name;
+      })
+      .filter((n) => !!n);
+  }
+
+  function setDetails() {
+    const names = getAssociatedNames(contractId);
+    const details = [
+      [
+        <td className="assets-table-item" key={`${contractId}-data`}>
+          Status:
+        </td>,
+        <td
+          className="assets-table-item"
+          style={{ flex: 4 }}
+          key={`${contractId}-data`}
+        >
+          <TransactionStatus
+            key={`${contractId}-confirmations`}
+            confirmations={confirmations}
+          />
+        </td>,
+      ],
+      [
+        <td className="assets-table-item" key={`${contractId}-data`}>
+          {`Associated Names (${names?.length}) :`}
+        </td>,
+        <td
+          className="assets-table-item"
+          style={{ flex: 4 }}
+          key={`${contractId}-data`}
+        >
+          {names
+            ? names.map((name) => (
+                <>
+                  <span
+                    className="assets-manage-button"
+                    key={`${contractId}-name`}
+                  >
+                    {name}
+                  </span>
+                  &nbsp;
+                </>
+              ))
+            : 'N/A'}
```

Comment:
<img width="1210" alt="image" src="https://user-images.githubusercontent.com/12800001/214640145-f85c36ac-54b9-4556-858f-a355f97d7031.png">
for now, can we do just a joined list of names, vs. the buttons
```suggestion
        <td className="assets-table-item" key={`${contractId}-data`}>
          {`Associated Names (${names?.length}) :`}
        </td>,
        <td
          className="assets-table-item"
          style={{ flex: 4 }}
          key={`${contractId}-data`}
        >
          {names.length ? names.join(', ') :  'N/A'}
```